docs/examples/simulation/ODE_example.py

```diff
@@ -0,0 +1,418 @@
+    if k == 33:  # model the turn left
+        new_state = turn_left.function(truth[k-1], time_interval=timestep_size, noise=False)
+        truth.append(GroundTruthState(
+            new_state,
+            timestamp=start_time + timedelta(seconds=timestep_size.total_seconds()*k)))
+
+    elif k == 66:  # model the turn right
+        new_state = turn_right.function(truth[k - 1], time_interval=timestep_size, noise=False)
+        truth.append(GroundTruthState(
+            new_state,
+            timestamp=start_time + timedelta(seconds=timestep_size.total_seconds()*k)))
+
+    else:  # straight trajectory
+        new_state = constant_velocity.function(
+            truth[k - 1], time_interval=timestep_size, noise=False)
+        truth.append(GroundTruthState(
+            new_state,
+            timestamp=start_time + timedelta(seconds=timestep_size.total_seconds()*k)))
+
+# Instantiate the measurement model
+measurement_model = LinearGaussian(ndim_state=5,
+                                   mapping=(0, 2),
+                                   noise_covar=np.array([[10, 0],
+                                                         [0, 10]]))
+
+# Collect the measurements and timestamps
+measurements, times = [], []
+for state in truth:
+    measurement = measurement_model.function(state, noise=True)
+    measurements.append(TrueDetection(
+        state_vector=measurement,
+        timestamp=state.timestamp,
+        groundtruth_path=truth,
+        measurement_model=measurement_model))
+
+    times.append(state.timestamp)
+
+# %%
+# Let's visualise the target ground truth and detections
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+from stonesoup.plotter import Plotterly
+
+plotter = Plotterly()
+plotter.plot_ground_truths(truth, [0, 2])
+plotter.plot_measurements(measurements, [0, 2])
+plotter.fig
+
+# %%
+# 2. Create the linearised function;
+# ----------------------------------
+# We have created a series of detections from a target moving with peculiar dynamics as can be seen
+# in the figure.
+# We can build the constant heading function and the class for the linearisation that would generate
+# a linearised transition model.
+#
+# We create here a copy of a standard Stone Soup transition model class, in which we perform the
+# automatic differentiation to obtain the ``jacobian`` matrix and perform the linearisation. In
+# addition, as can be seen in other transition models, we instantiate the methods to create a new
+# state given a previous one (``function``) and evaluate the covariance matrix (``covar``).
+
+
+# Create the constant heading function
+def constant_heading(state, **kwargs):
+    """
+        Function that describes the constant heading dynamics;
+        The states are a bit odd
+    """
+
+    (x, vx, s, vy, theta) = state
+    # generate the s component
+    s = torch.sqrt(vx*vx + vy*vy)
+    return s * torch.cos(theta), 0 * s, s * torch.sin(theta), 0. * s, 0. * theta
```

Comment:
```suggestion
    (x, vx, y, vy, theta) = state
    # generate the s component
    s = torch.sqrt(vx*vx + vy*vy)
    return s * torch.cos(theta), 0. * s, s * torch.sin(theta), 0. * s, 0. * theta
``` 
Just a small last thing that I noticed. Is the ``s`` taken from the state supposed to be a ``y``. (the variable isn't used so it doesn't change anything)